formats/json-okio/api/kotlinx-serialization-json-okio.api

```diff
@@ -0,0 +1,7 @@
+public final class kotlinx/serialization/json/okio/OkioStreamsKt {
+	public static final fun decodeFromOkio (Lkotlinx/serialization/json/Json;Lkotlinx/serialization/DeserializationStrategy;Lokio/BufferedSource;)Ljava/lang/Object;
+	public static final fun decodeOkioToSequence (Lkotlinx/serialization/json/Json;Lokio/BufferedSource;Lkotlinx/serialization/DeserializationStrategy;Lkotlinx/serialization/json/DecodeSequenceMode;)Lkotlin/sequences/Sequence;
+	public static synthetic fun decodeOkioToSequence$default (Lkotlinx/serialization/json/Json;Lokio/BufferedSource;Lkotlinx/serialization/DeserializationStrategy;Lkotlinx/serialization/json/DecodeSequenceMode;ILjava/lang/Object;)Lkotlin/sequences/Sequence;
+	public static final fun encodeToOkio (Lkotlinx/serialization/json/Json;Lkotlinx/serialization/SerializationStrategy;Ljava/lang/Object;Lokio/BufferedSink;)V
```

Comment:
`Sink.buffer()` is not a virtual function, its extension, therefore, for each encoding will be created new instance of `RealBufferedSink` and `Buffer`. The main impact turns out if someone passes `BufferedSink` (I think this will happen most often) - in this case, a second `Buffer` will be created in the new class, which uses an additional `Segment` - thereby we get memory overruns (two Segments when), extra allocations and a second useless copying of data in memory.

Alternatively, we can create several functions, one will take `BufferedSink`, the other a pure `Sink` and call first function with `.buffer()`.

Reply:
Can't we simply do `if (sink is BufferedSink) /*use it*/ else sink.buffer()`?

Reply:
I think we can add such a check, but in this case I think it should be explicitly specified in KDoc

Reply:
Yeah, why not